backend/src/controller/chatController.ts

```diff
@@ -373,10 +369,9 @@ function handleAddToChatHistory(req: OpenAiAddHistoryRequest, res: Response) {
 		req.session.levelState[level].chatHistory = pushMessageToHistory(
 			req.session.levelState[level].chatHistory,
 			{
-				completion: null,
 				chatMessageType,
 				infoMessage,
-			}
+			} as ChatMessage
```

Comment:
Is this method type safe? Does it need to be? Is it problematic if, because of client error, a message has superfluous properties?

Reply:
Hmm, good question. We might consider two improvements then:
- Create a type representing ChatMessages that originate from an API call (i.e. from the UI)
- Validate these messages on arrival, by checking their content conforms to our expected shape(s) and types.

Either or both of these could be done in a separate issue if you prefer, though you migth want to get them done now rather than kicking them down the road.

Reply:
Adding extra types seems not useful if types don't exist at runtime. We'd need to add explicit checks that would work even when the project is compiled to js, non?

Reply:
Okay, I've done my best shot at this, which may well could have been a separate PR's worth of work, so let me know if you want me to cherry pick commits across for ease of review.
- I've completely revamped the _models/chatMessage.ts_ file which defines the chat message types. All the chat message types which will (in the history) store the message in the `infoMessage` property are grouped together as `CHAT_MESSAGE_TYPE_AS_INFO`. This comes from an array of string literals. This same array is used to check that the message type in the request isn't bogus.
- While I was doing that I accidentally bumped into [this bug](https://github.com/ScottLogic/prompt-injection/issues/744) and solved it before realising there was already a ticket for it. I added an extra unit test to catch it.

this is a type-safe as I can get it, but it's kinda ugly and the names are confusing, so I'm happy to try another approach / take suggestions.